adds electron-photon Heads-on psl

szabo137 opened the pull request:
Adds `PhotonElectronHeadsOnSystem`, a phase space layout in which the electron and photon counter-propagate along a single axis.

AntonReinhard commented on src/phase_space_layouts/in_channel/two_body/heads_on.jl:
```diff
@@ -0,0 +1,47 @@
+        in_coords::NTuple{2, T}
+    ) where {T <: Real}
+    E, om = in_coords
+    rho = sqrt(E^2 - one(E))
```
This is a prime candidate for precision loss when E ~= 1, it should help to rewrite this as
```suggestion
    rho = sqrt((E - one(E)) * (E + one(E)))
```

szabo137 replied:
You are right, of course! I will consider this.

szabo137 replied:
I used this in the rewrite in the helper `_mag_from_E(E,m)`.

szabo137 commented on src/phase_space_layouts/in_channel/two_body/heads_on.jl:
```diff
@@ -0,0 +1,47 @@
+QEDbase.particle_direction(psl::PhotonElectronHeadsOnSystem) = psl.dir
+PhotonElectronHeadsOnSystem() = PhotonElectronHeadsOnSystem(ZAxis())
+
+#Base.broadcastable(psl::PhotonElectronHeadsOnSystem) = Ref(psl)
```
Because it is already implemented upstream for every `AbstractPhaseSpaceLayout`. I removed this line in the final version.

szabo137 commented:
@AntonReinhard This is now rewritten to accept any process with two incoming particles. Furthermore, I added a special heads-on layout, the center-of-momentum system. 

Feel free to complain about the implementation 😸 

Do you have an idea of how I could check at compile time that the process passed into `_build_momenta` has exactly two incoming particles? I assume you have some recurrive magic at hand 🤔

AntonReinhard commented:
> Do you have an idea of how I could check at compile time that the process passed into _build_momenta has exactly two incoming particles?

Shouldn't it be enough to just have an assert that `number_incoming_particles(proc) == 2`? That constant is compile time anyway.
